src/mono/wasi/Wasi.Build.Tests/WasiTemplateTests.cs

```diff
@@ -21,24 +21,126 @@ public WasiTemplateTests(ITestOutputHelper output, SharedBuildPerTestClassFixtur
+        string id = $"{config}_{GetRandomId()}";
+        string projectFile = CreateWasmTemplateProject(id, "wasiconsole");
+        string projectName = Path.GetFileNameWithoutExtension(projectFile);
+        File.Copy(Path.Combine(BuildEnvironment.TestAssetsPath, "SimpleMainWithArgs.cs"), Path.Combine(_projectDir!, "Program.cs"), true);
+
+        var buildArgs = new BuildArgs(projectName, config, aot, id, null);
+        buildArgs = ExpandBuildArgs(buildArgs);
+
+        string extraProperties = "";
+        if (aot)
+            extraProperties = "<RunAOTCompilation>true</RunAOTCompilation><_WasmDevel>false</_WasmDevel>";
+        if (singleFileBundle)
+            extraProperties += "<WasmSingleFileBundle>true</WasmSingleFileBundle>";
+        if (!string.IsNullOrEmpty(extraProperties))
+        AddItemsPropertiesToProject(projectFile, extraProperties);
+
+        BuildProject(buildArgs,
+                    id: id,
+                    new BuildProjectOptions(
+                        DotnetWasmFromRuntimePack: true,
+                        CreateProject: false,
+                        Publish: false,
+                        TargetFramework: BuildTestBase.DefaultTargetFramework));
+
+        // ActiveIssue: https://github.com/dotnet/runtime/issues/82515
+        int expectedExitCode = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? 1 : 42;
+
+        if (!_buildContext.TryGetBuildFor(buildArgs, out BuildProduct? product))
+            throw new XunitException($"Test bug: could not get the build product in the cache");
+
+        File.Move(product!.LogFile, Path.ChangeExtension(product.LogFile!, ".first.binlog"));
+
+        _testOutput.WriteLine($"{Environment.NewLine}Publishing with no changes ..{Environment.NewLine}");
+
+        bool expectRelinking = config == "Release";
+        BuildProject(buildArgs,
+                    id: id,
+                    new BuildProjectOptions(
+                        DotnetWasmFromRuntimePack: !expectRelinking,
+                        CreateProject: false,
+                        Publish: true,
+                        TargetFramework: BuildTestBase.DefaultTargetFramework,
+                        UseCache: false));
+    }
+
+    [Theory]
+    [InlineData("Debug", /*aot*/ false, /*singleFileBundle*/ false)]
+    [InlineData("Debug", /*aot*/ true, /*singleFileBundle*/ false)]
+    [InlineData("Release", /*aot*/ false, /*singleFileBundle*/ false)]
+    [InlineData("Release", /*aot*/ true, /*singleFileBundle*/ false)]
+    [InlineData("Debug", /*aot*/ false, /*singleFileBundle*/ true)]
+    // [InlineData("Debug", /*aot*/ true, /*singleFileBundle*/ true)] // https://github.com/dotnet/runtime/issues/95273
+    [InlineData("Release", /*aot*/ false, /*singleFileBundle*/ true)]
+    // [InlineData("Release", /*aot*/ true, /*singleFileBundle*/ true)] // https://github.com/dotnet/runtime/issues/95273
+    public void ConsoleBuildThenRunThenPublish(string config, bool aot, bool singleFileBundle)
```

Comment:
This is also same as `ConsoleBuildThenPublish`. Run the app in that test itself.

Reply:
No, it has `Run` while the other did not try to run. If we don't care about having tests of each part separately, then I will leave only `ConsoleBuildThenRunThenPublish` and remove the others.